.github/workflows/build-and-test.yml

```diff
@@ -25,6 +25,6 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install -r requirements.txt
```

Comment:
Not a big fan of the way requirement files are currently set up. Give this more thought